### PR TITLE
fix: jumpy review tx modal

### PR DESCRIPTION
**Pull request by DiogoSoaress**
## What it solves
- do not display nonce `0` while waiting for the estimations endpoint
- renders a Skeleton while waiting for the Simulation parameters

## How to test it
1. Go to a Safe and create a transaction
2. On the review modal, while the nonce is being retrieved, a Skeleton is displayed
3. In the same modal, a Skeleton component displays before the Simulation's accordion

## Screenshots
Screenshot 2022-09-26 at 15 33 17

### CODE REVIEW

File: src/components/tx/GasParams/index.tsx [hunk 1]
```diff
@@ -66,7 +69,7 @@ const GasParams = ({ params, isExecution, onEdit }: GasParamsProps): ReactElemen
         ) : (
           <Typography>
             Signing the transaction with nonce&nbsp;
-            {nonce !== undefined ? (
+            {nonce !== 0 || safeNonce === 0 ? (
```
Comment (katspaugh) [inline src/components/tx/GasParams/index.tsx:72]: 👀

File: src/components/tx/TxSimulation/index.tsx [hunk 2]
```diff
@@ -86,5 +86,15 @@ export const TxSimulation = (props: TxSimulationProps): ReactElement | null => {
     return null
   }
 
+  if (!props.transactions) {
+    return (
+      <Box sx={{ mt: 2 }}>
```
Comment (katspaugh) [inline src/components/tx/TxSimulation/index.tsx:91]: First of all, why do you need a box if you can just assign the margin to the skeletonWrapper? Secondly, for future: `<Box mt={2}`.

File: src/components/tx/GasParams/index.tsx (hunk 1 above)
Comment (katspaugh) [reply src/components/tx/GasParams/index.tsx:72]: I don't get this condition.

Comment (DiogoSoaress) [reply src/components/tx/GasParams/index.tsx:72]: I should have left a comment.... We were displaying 0 while waiting for the estimation. It only make sense to display the zero while waiting for the estimation if the `safeNonce` is indeed 0

File: src/components/tx/TxSimulation/index.tsx (hunk 2 above)
Comment (DiogoSoaress) [reply src/components/tx/TxSimulation/index.tsx:91]: I added the wrapper after the `<Box />`. thank you for the catch

File: src/components/tx/GasParams/index.tsx (hunk 1 above)
Comment (DiogoSoaress) [reply src/components/tx/GasParams/index.tsx:72]: on it ser! I was just testing

**Review by katspaugh — APPROVED** [review]
👍

**Comment by francovenica** [comment]
Is still jumpy, since there are many other fields besides the one for the estimation, there is the simulation and the tx data (this for safes 1/x) . Should we make a skeleton of each different block?
jumpy

**Comment by katspaugh** [comment]
The problem is we don't know ahead of time if the tx is executable (nonce can be not the earliest etc). So even if we show skeletons, they might disappear. So it will just jump in more cases than now.

**Comment by francovenica** [comment]
Fair enough.

LGTM. Can be merged when possible

**Pull request merged**